whisper-fine-tuning-event/run_eval_whisper_streaming.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+
+from transformers import pipeline
+from datasets import load_dataset, Audio
+import evaluate
+
+wer_metric = evaluate.load("wer")
+
+
+def is_target_text_in_range(ref):
+    if ref.strip() == "ignore time segment in scoring":
+        return False
+    else:
+        return ref.strip() != ""
+
+
+def get_text(sample):
+    if "text" in sample:
+        return sample["text"]
+    elif "sentence" in sample:
+        return sample["sentence"]
+    elif "normalized_text" in sample:
+        return sample["normalized_text"]
+    elif "transcript" in sample:
+        return sample["transcript"]
+    else:
+        raise ValueError(f"Sample: {sample.keys()} has no transcript.")
+
+
+def data(dataset):
+    for i, item in enumerate(dataset):
+        yield {**item["audio"], "reference": item["norm_text"]}
+
+
+def main(args):
+    batch_size = args.batch_size
+    whisper_asr = pipeline(
+        "automatic-speech-recognition", model=args.model_id, device=args.device
+    )
+    whisper_asr.model.config.max_length = 128
+
+    whisper_norm = whisper_asr.tokenizer._normalize
+
+    def normalise(batch):
+        batch["norm_text"] = whisper_norm(get_text(batch))
+        return batch
+
+    dataset = load_dataset(
+        args.dataset, args.config, split=args.split, streaming=True, use_auth_token=True
+    )
+
+    # Only uncomment for debugging
+    dataset = dataset.take(64)
```

Comment:
Likely hood is that participants won't dive deep into the contents of this file and are thus likely to miss this statement.

If it goes unnoticed, participants will only ever eval on the first 64 samples of data and will have partial eval results. IMO we should not set this by default. I suggest either:

1. We add an arg `max_eval_samples` that defaults to None. If specified, it controls the number of samples we evaluate on (`dataset.take(max_eval_samples)`)
2. We remove this line and only ever evaluate on the whole dataset

I'd be in favour of 1, but strongly believe that we can't leave in evaluating on only 64 samples by default

Reply:
Good idea. I've added #1 approach in the script. Works fine! Thanks!